### PR TITLE
Feature/associations

### PR DESCRIPTION
**Pull request by Buyaki01**
- Create a class Classroom with the following: @label instance variable, should be initialized in the constructor. Setter and getter for @label (remember about attr_accessor).
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented: Create the has-many side (a classroom has many students). Create the belongs-to side (a student belongs to a classroom). Make sure that when adding a student to a classroom it also sets the classroom for the student. Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class Book with the following: @title and @author instance variables, should be initialized in the constructor. Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following: @date instance variable, should be initialized in the constructor. Setter and getter for @date (remember about attr_accessor).
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented: Create the has-many side of Book and Rental (a book has many rentals). Create the belongs-to side of Rental and Book (a rental belongs to a book). Create the has-many side of Person and Rental (a person has many rentals). Create the belongs-to side of Rental and Person (a rental belongs to a person). Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './rental'
+
+class Book
```
Comment (omar-labana) [inline book.rb:3]: - [ ] A method for adding rentals to `@rentals` should be present in order to make this class functional, please add it. I'd suggest taking a look at [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship).

File: person.rb [hunk 2]
```diff
@@ -1,20 +1,31 @@
...
+class Person < Nameable
+  attr_accessor :name, :age
+  attr_reader :id, :rental
+
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rental = []
```
Comment (omar-labana) [inline person.rb:13]: - [ ] Similar to the book case, a person can have many rentals but the function `add_rental` to this attribute is missing, please add it. I'd suggest taking a look at [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship).

**Review by omar-labana — COMMENTED** [review]
Hi @Buyaki01, 

Excellent work so far! 👌
You have implemented almost all the required features correctly and without any bugs or errors. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
_N/A_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by o-rashad — APPROVED** [review]
Asslam Alaikum @Buyaki01 ,😊

Hi👋🏽, 
Your project is complete!  
Thanks For applying the changes requested by the previous reviewer 👌 
There is nothing else to say other than... it's time to merge it 😎 

# Congratulations! 🎉 

Happy coding!👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**